src/plone/restapi/services/aliases/add.py

```diff
@@ -84,9 +84,48 @@ class AliasesRootPost(Service):
     """Creates new aliases via controlpanel"""
 
     def reply(self):
-        data = json_body(self.request)
         storage = getUtility(IRedirectionStorage)
-        aliases = data.get("items", [])
+
+        if self.request.getHeader("Content-Type") == "text/csv":
```

Comment:
I don't think this and the next `if` statment should be the only logical checks for a CSV file upload. See https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/controlpanel/browser/redirects.py#L380 for validation of a CSV file.